tests/service/SyntaxTreeTests.fs

```diff
@@ -175,17 +175,24 @@ let ParseFile fileName =
             $"%A{ast}\n\n%s{diagnostics}"
         |> normalize
         |> sprintf "%s\n"
+
     let bslPath = $"{fullPath}.bsl"
+    let tmpPath = $"{fullPath}.tmp"
 
     let expected =
         if File.Exists bslPath then
             File.ReadAllText bslPath |> normalize
         else
             "No baseline was found"
 
+    let equals = expected = actual
     let testUpdateBSLEnv = System.Environment.GetEnvironmentVariable("TEST_UPDATE_BSL")
 
     if not (isNull testUpdateBSLEnv) && testUpdateBSLEnv.Trim() = "1" then
         File.WriteAllText(bslPath, actual)
+    elif not equals then
+        File.WriteAllText(tmpPath, actual)
+    else
+        File.Delete(tmpPath)
```

Comment:
Would be good to avoid IO-attempt in the happy path

Reply:
Do you think it is something that would make a measurable difference in these tests?

Reply:
These things tend to accummulate over time with more tests and more functinalities in the suite.
If it can be easily avoided, we shouldn't be doing unnecessary IO when tests are running.

Reply:
It looks like these tests access IO enough already for it not to matter much here. It simplifies the work with the parser, as it's easy to see what got changed without requiring to clean this files manually later.

Reply:
Or I could, for example, check some environment variable indicating a CI run if you share a pointer

Reply:
If the TEST_UPDATE_BSL is not appropriate, see here what get's configured for CI runs:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=241480&view=logs&jobId=b4ab8557-6e24-5e79-3313-9df489728e07&j=b4ab8557-6e24-5e79-3313-9df489728e07&t=cdc24b7f-1874-5a89-2ab2-501a708f5d36

e.g. "BUILD_REASON" is set for CI runs.